content/4_Gold/Tree_Euler.mdx

```diff
@@ -542,6 +563,86 @@ public class Main {
 
 </JavaSection>
 
+<PySection>
+
+```py
+import sys
+
+sys.setrecursionlimit(10**9)
+
+NMAX = 200005
+n, q = map(int, input().split())
+
+v = list(map(int, input().split()))
+
+
+class SegmentTree:
+	def __init__(self, arr: list[int], n: int) -> None:
```

Comment:
use `List` from `typing`

Reply:
What do you mean ? Isn't the type hint right ?

Reply:
I used list

Reply:
generic type hints aren't supported till 3.9 i think, and usaco uses 3.6
so just `from typing import List` and use that instead

Reply:
![image](https://github.com/cpinitiative/usaco-guide/assets/59292725/cc5c5875-4f8c-451b-9149-ad5d48eb89b5)
It's supported since 3.5

Reply:
Problem solved :))